### ISSUE TITLE
SIGs need a Roster on who is on a SIG

### ISSUE THREAD

**Issue by dims** [kubernetes/community#5854]
Having a Roster of some kind:
- will improve morale, have a sense of belonging
- A chance to celebrate
- Identify themselves with the group
- Natural mechanism to figure out who can vote
- Pool of folks SIG chairs and TLs can rely on
- Folks who are not in the roster can aspire to get there (part of a contributor ladder)
- Other SIGs (and general folks) will know who to reach out to (other than chairs and TLs)
- Possibly even improve stale PRs and Issues (or project boards) if we give the responsibility of triage to those on the roster.

Currently it's a mix of those in OWNER_ALIASES, google groups etc and it is hard to even run an election if we don't know who gets to vote. Looking up devstats for contributions to a SIG is dicey as well as it leaves out a bunch of folks doing important work (NOT in github).

So we need to standardize a mechanism for SIGs to curate/update a roster. I would suggest that SIG-Chairs are responsible to keep it up-to-date and to invite (or approve) people to be on the roster. Personally, i'd lean towards a github group backed by a yaml (with sig-chairs as approvers) which is periodically vetted both for stale entries as well as a conscious backlog of folks to promote to the roster as well.

**Comment by jberkus** [comment]
My big concern about this would be staleness, given that we're not good about keeping even the lists of subproject owners up to date.

**Comment by ehashman** [comment]
Re: staleness, is this something that could be lumped into the annual report work? We used that as an opportunity to check OWNERS files as well

**Comment by ehashman** [comment]
N.B. each SIG could have the option of maintaining this in k/org as GitHub teams, see e.g. https://github.com/kubernetes/org/blob/576e433048f585039d3c5b7a1cd4de7213170446/config/kubernetes/sig-instrumentation/teams.yaml#L20

A lot of our GH teams are quite out of date.

**Comment by tengqm** [comment]
+1

Maybe a related change is to assign some roles inside each SIG, e.g., a security focal, a documentation liaison etc. This will help the docs team identify the person who can help provide technical lgtms.

**Comment by sftim** [comment]
I like the idea of having that roster be (optionally) mirrored to a team in GitHub's API.

**Comment by BenTheElder** [comment]
This needs to not be arbitrary / per-sig if we're going to use it to vote for SIG leadership.

I'm not sure it should be used for that particular purpose though https://github.com/kubernetes/community/issues/5855#issuecomment-866497734

**Comment by saschagrunert** [comment]
I generally like the proposal. Speaking for SIG Release we already maintain something similar for each cycle as well as in the release engineering subproject.

**Comment by neolit123** [comment]
-1 too much work for distributed SIGs like Cluster Lifecycle and maybe Cloud Provider (not speaking for them).

@jberkus 

> My big concern about this would be staleness, given that we're not good about keeping even the lists of subproject owners up to date.

i agree. this feels like yet another administrative complication that from my perspective, gains no benefits in a best effort community such as k8s.

**Comment by lavalamp** [comment]
A roster seems mandatory if there is to be a vote. However a roster requires maintenance and objective criteria for who can be on it. Administrating this can be very expensive in terms of human effort.

There would have to be some requirement on SIG members, and I think it would easy for a well-intentioned system to in practice be exclusionary and make it harder for people to join the project.

**Comment by jberkus** [comment]
Could we make this semi-automated somehow?  That is, Org Member + 10 contributions == automatic invite?

**Comment by lavalamp** [comment]
I think something like that would be best, but the codebase requires some curation. Some SIGs co-own some sections of code (that's fine, you get invites to both) and some code is unclearly owned. The major problem is that the code is just not clearly enough delineated; e.g. the auto-SIG-labeling code gives my SIG a lot of false positives.

I think a backup mechanism would be needed to let SIGs grant membership to people who are performing non-code tasks. E.g. many chairs might be missed by such a mechanism!

**Comment by PushkarJ** [comment]
Not claiming that this is perfect or something we should adopt but in terms of exploring prior art as it relates to roster, membership criteria and role definitions, CNCF TAG Security has one: https://github.com/cncf/tag-security/blob/main/governance/roles.md

Also, +1 on automating some of it using Github teams

**Comment by jberkus** [comment]
@lavalamp yeah, I'm just looking for "how do we make this so easy for SIG chairs that they can do it every week".  Because if it's not that easy, it won't happen at all.
